flow/project.py

```diff
@@ -1692,18 +1715,86 @@ def _store_bundled(self, operations):
         else:
             h = '.'.join(op.id for op in operations)
             bid = '{}/bundle/{}'.format(self, sha1(h.encode('utf-8')).hexdigest())
-            fn_bundle = self._fn_bundle(bid)
+            fn_bundle = self._fn_stored('.bundles', bid)
             os.makedirs(os.path.dirname(fn_bundle), exist_ok=True)
             with open(fn_bundle, 'w') as file:
                 for operation in operations:
                     file.write(operation.id + '\n')
             return bid
 
+    def _store_aggregates(self, operations):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param operations:
+            The operations to be submitted.
+        :type operations:
+            A sequence of instances of :py:class:`.JobOperation`
+        """
+        for operation in operations:
+            op_fn = '{}.txt'.format(operation.name)
+            aggregate_wid = '{} {} \n'.format(operation.id, ' '.join(map(str, operation._jobs)))
+            fn_aggregate = self._fn_stored('.aggregates', op_fn)
+            os.makedirs(os.path.dirname(fn_aggregate), exist_ok=True)
+            if os.path.exists(fn_aggregate):
+                with open(fn_aggregate, 'r+') as file:
+                    agg_file_contents = file.read()
+                    if aggregate_wid in agg_file_contents:
+                        continue
+                    else:
+                        file.write(aggregate_wid)
+            else:
+                with open(fn_aggregate, 'w') as file:
+                    file.write(aggregate_wid)
+
+    def _fetch_aggregates(self, group):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param group:
+            FlowGroup associated with the operation.
+        :type group:
+            list :py:class:`flow.FlowGroup`
+        :return:
+            Fetched signac job handles that were previously submitted.
+        :rtype:
+            tuple
+        """
+        fetched_aggregates = []
```

Comment:
Why not make this a generator function?

Reply:
Yes, we could do that. Thank you for the suggestion.